### PR TITLE
coreos-meta-translator: add digitalocean

### PR DESCRIPTION
**Pull request by bgilbert**
Depends on https://github.com/coreos/fedora-coreos-stream-generator/pull/9.

### CODE REVIEW

File: coreos-meta-translator/trans.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def artifact(i):
         # build the architectures dict
         arch_dict = {"media": {}}
         ensure_dup(input_, arch_dict, "ostree-commit", "commit")
-        generic_arches = ["aliyun", "aws", "azure", "exoscale", "qemu", "metal", "openstack", "vmware", "gcp"]
+        generic_arches = ["aliyun", "aws", "azure", "exoscale", "qemu", "metal", "openstack", "vmware", "gcp", "digitalocean"]
```
Comment (dustymabe) [inline coreos-meta-translator/trans.py]: nit, looks like we were trying to go alphabetical originally, maybe we go back to that?

**Comment by bgilbert** [comment]
Sorted the platform list, and also renamed a couple variables for clarity.

**Comment by dustymabe** [comment]
ha nice. I was actually thinking to myself that "arch" was misleading, but didn't want to ask you to change that in this PR. Nice work!

**Comment by lucab** [comment]
Uh, wait, this introduced a split brain definition of the `digitalocean` platform ID:
 * for Afterburn, it means a platform without DHCP requiring additional network bringup
 * for coreos-assembler, it means a platform with DHCP and no further special requirements

I fear this is going to be a blocker for providing first-class DO images (assuming they still need network bringup). I'm still reading through all the linked issues/PRs but I havent's seen yet those details of our plans. What is our strategy out of this?

**Comment by bgilbert** [comment]
@lucab I don't think there's a conflict.  Afterburn runs successfully on `custom_ipxe` systems, and produces network information that matches DHCP.  So Afterburn's definition is fine, and once it supports driving NM, that can happen unconditionally.

**Comment by lucab** [comment]
From out-of-band discussion: we are not totally sure, but this is likely going to be fine. If not, we'll come up with some additional way to let us do auto-detection here.

**Pull request merged**